oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcUserService.java

```diff
@@ -173,8 +173,15 @@ private boolean shouldRetrieveUserInfo(OidcUserRequest userRequest) {
 				.equals(userRequest.getClientRegistration().getAuthorizationGrantType())) {
 			// Return true if there is at least one match between the authorized scope(s)
 			// and accessible scope(s)
+			// As per spec, in Section 5.1 Successful Access Token Response
+			// https://www.rfc-editor.org/rfc/rfc6749#section-5.1
```

Comment:
This comment does not make sense in this context. I think we should be more specific.